File: Trabalhos Práticos/ArquivoTP.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    double x;

    FILE *arq = fopen("ArqEntrada.txt", "w"); // Abertura do arquivo de entrada //

    // Leitura dos números e escrita no arquivo //
    for (int i = 0; i < n; i++) {
        scanf("%lf", &x);
        fprintf(arq, "%.9lf\n", x); 
    }

    fclose(arq);

    FILE *arq1 = fopen("ArqEntrada.txt", "r");

    // Armazenar os números em um array
    double numeros[n];
    int i = 0;
    while (i < n && fscanf(arq1, "%lf", &numeros[i]) == 1) {
        i++;
    }

    fclose(arq1);

    // Impressão dos números invertidos
    for (int j = i - 1; j >= 0; j--) {
        if (numeros[j] == (int)numeros[j]) {
            printf("%.0lf\n", numeros[j]);
        } else {
            printf("%.9g\n", numeros[j]); 
        }
    }

    return 0;
}

File: Trabalhos Práticos/ArquivoTP.java
import java.io.RandomAccessFile;

public class ArquivoTP {
    public static void main (String[] args) throws Exception {
        double x;
        int n = MyIO.readInt();

        RandomAccessFile arq = new RandomAccessFile("ArqEntrada.txt", "rw"); // Abertura do arquivo //

        // Leitura dos numeros //
        for (int i = 0; i < n; i++) {
            x = MyIO.readDouble();
            arq.writeDouble(x);
        }

        arq.close();

        RandomAccessFile arq1 = new RandomAccessFile("ArqEntrada.txt", "rw"); // Abrindo novamente o arquivo //

        double y;

        // Impressao dos numeros invertidos //
        for (int i = 0; i < n; i++) {
            arq1.seek((n - 1 - i)*8);
            y = arq1.readDouble();
            if (y % 1 != 0) {
                MyIO.println(y);
            } else {
                MyIO.println((int)y);
            }
        }

        arq1.close();   
    }
}
